src/chigame/users/views.py

```diff
@@ -120,3 +120,11 @@ def cancel_friend_invitation(request, pk):
     else:
         messages.error(request, "Something went wrong please try again later!")
     return redirect(reverse("users:user-profile", kwargs={"pk": request.user.pk}))
+
+
+@login_required
+def user_inbox_view(request, pk):
```

Comment:
You should add some sort of error handling, so that the current user can only see their own inbox. I tested with objects user1, user2, and a notification sent to user1. When I'm logged in as user2, I can't see anything in inbox. But when logged in as user1, I see the same notification at "users/inbox/pk" for any pk. This would be fixed simply by limiting to only being able to view the user's own inbox.

Reply:
Modified!

Reply:
I think it would make more sense to just redirect back to your page and also have a message that says "Not your inbox". When I try to access inboxes of users that haven't been created yet, it still gives the same error, but these users shouldn't have an inbox page at all.